### PR TITLE
Remove onClickListener from the GroupAdapter constructor

### PR DESCRIPTION
**Pull request by Teovald**
Hi, 

I have started playing with Groupie and so far I really like this lib.  
One thing that bothers me is that if I don't want my items to be clickable, I still need to provide an
onClickListener to the GroupAdapter and to override isClickable to false (so you also don't get 'click' sounds for unclickable elements).  

I think it would be easier to just drop this feature, it is easy to add click listener wherever you want in the item hierarchy with databinding anyway.

### CODE REVIEW

File: example/src/main/java/com/genius/groupie/example/MainActivity.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
         gray = ContextCompat.getColor(this, R.color.background);
         betweenPadding = getResources().getDimensionPixelSize(R.dimen.padding_small);
 
-        groupAdapter = new GroupAdapter(this);
+        groupAdapter = new GroupAdapter();
```
Comment (lisawray) [inline example/src/main/java/com/genius/groupie/example/MainActivity.java]: I'd like to keep the possibility of a top level click listener, as this is a pattern many people have continued from ListView.  However, I agree it's a good idea not to require it!  

Can you add `GroupAdapter.setClickListener` as a replacement?

File: example/src/main/java/com/genius/groupie/example/item/CardItem.java [hunk 2]
```diff
@@ -3,13 +3,13 @@
 import android.support.annotation.ColorRes;
 
 import com.genius.groupie.Item;
-import com.genius.groupie.example.MainActivity;
 import com.genius.groupie.example.databinding.ItemCardBinding;
 
 import java.util.HashMap;
 import java.util.Map;
 
-import static com.genius.groupie.example.MainActivity.*;
```
Comment (lisawray) [inline example/src/main/java/com/genius/groupie/example/item/CardItem.java]: Looks like this file shouldn't be changed

File: groupie/src/main/java/com/genius/groupie/GroupAdapter.java [hunk 3]
```diff
@@ -53,13 +50,12 @@ public int getSpanCount() {
     }
 
     @Override public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
-        Item contentItem = getItem(position);
-        contentItem.bind(holder, position, onItemClickListener);
```
Comment (lisawray) [inline groupie/src/main/java/com/genius/groupie/GroupAdapter.java]: There should still be a generic `onBind()` without payload.  Why is this removed?

File: groupie/src/main/java/com/genius/groupie/Item.java [hunk 4]
```diff
@@ -23,28 +22,14 @@
 
     protected GroupDataObserver parentDataObserver;
 
-    public void bind(RecyclerView.ViewHolder viewHolder, int position, View.OnClickListener onItemClickListener) {
```
Comment (lisawray) [inline groupie/src/main/java/com/genius/groupie/Item.java]: Why is this removed?

File: groupie/src/main/java/com/genius/groupie/Item.java [hunk 5]
```diff
@@ -93,13 +78,11 @@ public void bind(T viewBinding, int position, List<Object> payloads) {
         return this == item ? 0 : -1;
     }
 
-    public boolean isClickable() {
```
Comment (lisawray) [inline groupie/src/main/java/com/genius/groupie/Item.java]: I'd like to see this stay, as it doesn't matter unless the click listener is set.

File: example/src/main/java/com/genius/groupie/example/item/CardItem.java (hunk 2 above)
Comment (Teovald) [inline example/src/main/java/com/genius/groupie/example/item/CardItem.java]: yes, looks like auto formatting. benign anyway but I can remove it.

Comment (lisawray) [inline example/src/main/java/com/genius/groupie/example/item/CardItem.java]: It just messes up git blame. :)  I hope to get a style check going one of these days!

Comment (Teovald) [inline example/src/main/java/com/genius/groupie/example/item/CardItem.java]: no worries, I will revert this file :-)

File: example/src/main/java/com/genius/groupie/example/MainActivity.java (hunk 1 above)
Comment (Teovald) [inline example/src/main/java/com/genius/groupie/example/MainActivity.java]: I still have nightmares because of some of the bugs caused by this ItemClickListener :D  , it was so bad !  
If you think we should keep it though, yeah just having it as a setter sounds like a good compromise.

File: groupie/src/main/java/com/genius/groupie/GroupAdapter.java (hunk 3 above)
Comment (Teovald) [inline groupie/src/main/java/com/genius/groupie/GroupAdapter.java]: I should have done this in another commit  : I removed it because it is never going to be called.  
If you override the onBind with payload, the classical one is useless.

Comment (lisawray) [inline groupie/src/main/java/com/genius/groupie/GroupAdapter.java]: another commit sounds good with a test or two, I will add an issue to track this :)

File: groupie/src/main/java/com/genius/groupie/Item.java (hunk 5 above)
Comment (Teovald) [inline groupie/src/main/java/com/genius/groupie/Item.java]: Ok, what do you think the default should be ?  
I found it very perturbing to have items clickable by default, but maybe it is just me

Comment (lisawray) [inline groupie/src/main/java/com/genius/groupie/Item.java]: I think true is fine.  If you set the clicklistener then you would expect your items to be clickable by default, I think.  If you do not then it doesn't matter!
